### PR TITLE
ARROW-4923: [Java] Add methods to set long value at given index in DecimalVector

### PR DESCRIPTION
**Pull request by pprudhvi**

### CODE REVIEW

File: java/vector/src/main/java/org/apache/arrow/vector/util/DecimalUtility.java [hunk 1]
```diff
@@ -204,6 +205,16 @@ public static void writeBigDecimalToArrowBuf(BigDecimal value, ArrowBuf bytebuf,
     writeByteArrayToArrowBuf(bytes, bytebuf, index, padValue);
   }
 
+  /**
+   * Write the given long to the ArrowBuf at the given value index.
+   */
+  public static void writeLongToArrowBuf(long value, ArrowBuf bytebuf, int index) {
+    final long startIndex = bytebuf.memoryAddress() + index * DECIMAL_BYTE_LENGTH;
```
Comment (praveenbingo) [inline java/vector/src/main/java/org/apache/arrow/vector/util/DecimalUtility.java]: addressOfValue might be a better name..

File: java/vector/src/test/java/org/apache/arrow/vector/TestDecimalVector.java [hunk 2]
```diff
@@ -161,6 +161,34 @@ public void testWriteBigEndian() {
     }
   }
 
+  @Test
+  public void testLongReadWrite() {
+    try (DecimalVector decimalVector = TestUtils.newVector(DecimalVector.class, "decimal",
+            new ArrowType.Decimal(38, 0), allocator)) {
+      decimalVector.allocateNew();
+      long long1 = 0L;
```
Comment (praveenbingo) [inline java/vector/src/test/java/org/apache/arrow/vector/TestDecimalVector.java]: can be simplified by storing the values in an array and iterating over it to do the set and assert..

**Review by praveenbingo — APPROVED** [review]
+1 LGTM.

Thanks @pprudhvi

File: java/vector/src/test/java/org/apache/arrow/vector/TestDecimalVector.java [hunk 3]
```diff
@@ -161,6 +161,29 @@ public void testWriteBigEndian() {
...
+
+      for (int i = 0; i < longValues.length; ++i) {
+        decimalVector.set(i, longValues[i]);
+      }
+
+      decimalVector.setValueCount(8);
+      assertEquals(8, decimalVector.getValueCount());
+
+      for (int i = 0; i < longValues.length; ++i) {
+        decimalVector.set(i, longValues[i]);
```
Comment (pravindra) [inline java/vector/src/test/java/org/apache/arrow/vector/TestDecimalVector.java]: why is there a second set here ?

**Review by pravindra — APPROVED** [review]

**Pull request closed without merging**